src/lib/docker-utils.coffee

```diff
@@ -21,13 +22,23 @@ applyDelta = (imgSrc, deltaUrl, applyTimeout, opts, onProgress) ->
 			else if parseInt(res.headers['content-length']) is 0
 				reject(new Error('Invalid delta URL.'))
 			else
-				deltaStream = dockerDelta.applyDelta(imgSrc, timeout: applyTimeout)
+				deltaStream = dockerDelta.applyDelta(imgSrc, { log, timeout: applyTimeout })
 				res.pipe(deltaStream)
 				.on('id', (id) -> resolve('sha256:' + id))
 				.on 'error', (err) ->
-					req.abort()
+					log("Delta stream emitted error: #{err}")
+					req.abort(err)
 					reject(err)
 
+applyBalenaDelta = (deltaImg, token, onProgress, log) ->
+	log('Applying balena delta...')
+	if token?
+		log('Using registry auth token')
+		auth = { authconfig: registrytoken: token }
+	@dockerProgress.pull(deltaImg, onProgress, auth)
```

Comment:
Now that deltas have tags, I wonder if `_removeImageIfNotNeeded` in src/compose/images.coffee will keep working well... I think it should be fine, but we'll have to test that the supervisor cleans up old images well when doing balena delta updates.